src/streaming/text/TextTracks.js

```diff
@@ -503,15 +503,19 @@ function TextTracks(config) {
                                 track.addCue(cue);
                             }
                         }
-
                     }
+
+                    // Remove old cues
+                    const bufferToKeep = settings.get().streaming.buffer.bufferToKeep;
+                    const currentTime = videoModel.getTime();
+                    _deleteOutdatedTrackCues(track, currentTime - bufferToKeep, currentTime);
```

Comment:
[Line 772](https://github.com/Dash-Industry-Forum/dash.js/pull/4305/files#diff-033321ac6c729cfa349f6ca06067679dc6919fe8fe1023ec6fd990c83e98ac9aR772):

```
function _deleteOutdatedTrackCues(track, start, end) {
```
 
Suggests `currentTime - bufferToKeep` is being used for `start` and `currentTime` is being used for `end`. That seems wrong: shouldn't `currentTime - bufferToKeep` be the `end` and 0 be the `start`, if `(currentTime - bufferToKeep) > 0`?

Otherwise the code reads as though the cues right up until just before the bleeding edge are being deleted, and anything somehow older than `bufferToKeep` is kept forever.

Reply:
Thanks, I agree the range should be from `0` to `currentTime - bufferToKeep`